src/index.js

```diff
@@ -1,40 +1,87 @@
 const list = document.getElementById('list');
-const position = 'beforeend';
-
-const toDoList = [
-  {
-    index: 0,
-    description: 'Reading books',
-    completed: true,
-  },
-  {
-    index: 1,
-    description: 'Going for a walk',
-    completed: true,
-  },
-  {
-    index: 2,
-    description: 'Meet families',
-    completed: true,
-  },
-  {
-    index: 3,
-    description: 'play football',
-    completed: true,
-  },
-];
-
-const todo = () => {
-  for (let i = 0; i < toDoList.length; i += 1) {
-    const item = `<li class="item">
+const input = document.getElementById('input');
+const addBtn = document.querySelector('.add');
+
+let LIST;
+let id;
+
+const check = 'fa-square-check';
+const uncheck = 'fa-square';
+const line = 'line-through';
+// local storage
+
+function addToDo(todo, id, done, trash) {
+  if (trash) { return; }
+  const Done = done ? check : uncheck;
+  const Line = done ? line : '';
+  const position = 'beforeend';
+
+  const item = `<li class="item">
               <div class="left">
-              <i class="fa-regular fa-square"></i>
-              <p class="text">${toDoList[i].description}</p>
+              <i class="fa-regular ${Done}" job="complete" id=${id}></i>
+              <p class="text ${Line}">${todo}</p>
               </div>
-              <i class="fa-solid fa-trash" class="complete"></i>
+              <i class="fa-solid fa-trash" job="delete" id=${id}></i>
             </li>`;
-    list.insertAdjacentHTML(position, item);
-  }
+
+  list.insertAdjacentHTML(position, item);
+}
+
+const loadList = (array) => {
+  array.forEach((item) => {
+    addToDo(item.todo, item.id, item.done, item.trash);
+  });
 };
+const myData = localStorage.getItem('TODO');
+if (myData) {
+  LIST = JSON.parse(myData);
+  id = LIST.length;
+  loadList(LIST);
+} else {
+  LIST = [];
+  id = 0;
+}
+
+addBtn.addEventListener('click', () => {
+  const todo = input.value;
+
+  if (todo) {
+    addToDo(todo, id, false, false);
+    LIST.push({
+      todo,
+      id,
+      done: false,
+      trash: false,
+    });
+
+    id += 1;
+    localStorage.setItem('TODO', JSON.stringify(LIST));
+  }
+
+  input.value = '';
+});
+
+function completeToDo(element) {
+  element.classList.toggle(check);
+  element.classList.toggle(uncheck);
+  element.parentNode.querySelector('.text').classList.toggle(line);
+  LIST[element.id].done = !LIST[element.id].done;
+}
+
+function removeToDo(element) {
+  element.parentNode.parentNode.removeChild(element.parentNode);
+  LIST[element.id].trash = true;
+}
```

Comment:
- [ ] Please make sure deleting a task should remove it from the local storage and update all remaining items' indexes/ids, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1) as requested in the project requirements. Currently, It's not been removed from the local storage you are handling the removed tasks with the trash field.